### PR TITLE
Degree model and tests

### PR DESCRIPTION
**Pull request by DanielItkes**
Model, test data and tests for the Degree application.

### CODE REVIEW

**Review by shakedhagag — COMMENTED** [review]
Reviewed.

**Review by DanWaldman — APPROVED** [review]

**Review by razashtar93 — APPROVED** [review]

**Comment by DanWaldman** [comment]
Resolved #48

**Comment by Omeramsc** [comment]
> Resolved #48 #46

This should be included in (an informative...) PR message, not as a comment. When its a comment it does not actually link the issue to the PR and does not resolve it.

**Comment by Yarboa** [comment]
> > Resolved #48 #46
> 
> This should be included in the PR message, not as a comment. When its a comment it does not actually link the issue to the PR and does not resolve it.
> 
> Also, I see a giant PR with a lot of things added and a lot of things removed; please write an informative PR message, otherwise it is really hard to review.

+1, 
Please try to move the Degree and Course to different Model file.
In case you encounter issues, please contact in #academe slack

**Comment by Yarboa** [comment]
> > > Resolved #48 #46
> > 
> > 
> > This should be included in the PR message, not as a comment. When its a comment it does not actually link the issue to the PR and does not resolve it.
> > Also, I see a giant PR with a lot of things added and a lot of things removed; please write an informative PR message, otherwise it is really hard to review.
> 
> +1, Please try to move the Degree and Course to different Model file. In case you encounter issues, please contact in #academe slack

@DanielItkes 

Please refer this file remove the Coarse from model.py and create new model_<name>.py 
https://pastebin.com/tBCsDi7u

do not forget to makemigrats again

**Comment by DanielItkes** [comment]
@Yarboa Modified PR so it only contains Degree model, test data and tests so it would be smaller and more understandable.

**Comment by DanielItkes** [comment]
> > > > Resolved #48 #46
> > > 
> > > 
> > > This should be included in the PR message, not as a comment. When its a comment it does not actually link the issue to the PR and does not resolve it.
> > > Also, I see a giant PR with a lot of things added and a lot of things removed; please write an informative PR message, otherwise it is really hard to review.
> > 
> > 
> > +1, Please try to move the Degree and Course to different Model file. In case you encounter issues, please contact in #academe slack
> 
> @DanielItkes
> 
> Please refer this file remove the Coarse from model.py and create new model_.py https://pastebin.com/tBCsDi7u
> 
> do not forget to makemigrats again

ACK. Removed Course model related stuff to make PR smaller and more readable.

**Comment by DanielItkes** [comment]
> > > Resolved #48 #46
> > 
> > 
> > This should be included in the PR message, not as a comment. When its a comment it does not actually link the issue to the PR and does not resolve it.
> > Also, I see a giant PR with a lot of things added and a lot of things removed; please write an informative PR message, otherwise it is really hard to review.
> 
> +1, Please try to move the Degree and Course to different Model file. In case you encounter issues, please contact in #academe slack

ACK. Removed Course related code to make PR smaller and more readable.

**Comment by DanielItkes** [comment]
> > Resolved #48 #46
> 
> This should be included in the PR message, not as a comment. When its a comment it does not actually link the issue to the PR and does not resolve it.
> 
> Also, I see a giant PR with a lot of things added and a lot of things removed; please write an informative PR message, otherwise it is really hard to review.

ACK. Removed Course related code to make PR smaller and more readable.

File: AcadeMeData/tests.py [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import pytest
-from AcadeMeData.models import User, Professor, University
+from AcadeMeData.models import User, Professor, University, Degree
```
Comment (Yarboa) [inline AcadeMeData/tests.py:2]: Please move this to separated file

File: AcadeMeData/tests.py [hunk 2]
```diff
@@ -47,51 +35,75 @@ def test_get_user(self):
...
+                              uni_name='Reichman University', location="Herzlia",
+                              uni_description="A nice place"):
+        university = TestUniversityModel.generate_university(self, university_id, uni_name, location, uni_description)
+        professor_for_example = self.generate_professor(professor_id, name, university, description, rate)
+        professor = Professor.get_professor(name)
         assert professor.get_name() == professor_for_example.name
         assert professor.get_description() == professor_for_example.description
+
+
+@pytest.mark.django_db
```
[the remark this answers is not in the document]
Comment (Yarboa) [reply AcadeMeData/tests.py:93]: And add the following

import pytest
from AcadeMeData.models import Degree

File: AcadeMeData/tests.py (hunk 1 above)
Comment (DanielItkes) [reply AcadeMeData/tests.py]: Done.

File: AcadeMeData/tests.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (DanielItkes) [reply AcadeMeData/tests.py]: Done.

File: AcadeMeData/test_degree.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
+import pytest
+from AcadeMeData.models import Degree
+
+
+@pytest.mark.django_db
+class TestDegreeModel:
+    def degree_example(self):
```
Comment (Yarboa) [inline AcadeMeData/test_degree.py:7]: Please use fixture here
please refer the link below:

Consider open an Issue to make all fixtures in one file for laater work

https://github.com/redhat-beyond/AcadeMe/pull/66/commits/eb91a64b9f84f5a2a58202e63a39680c3f153490#diff-10c42fcf224bda15e60cdea601bebe0bbe71f0a4a5930d09dfac6243707b9933R95

Comment (DanielItkes) [reply AcadeMeData/test_degree.py]: Done

File: AcadeMeData/test_degree.py [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    @pytest.fixture
+    def generate_degree(self, degree_id=1, name='History', universities="Ben Gurion University, Reichman University",
+                        description="Learn about historic events and their influences on the world"):
+        degree = Degree(degree_id=degree_id, name=name, universities=universities,
+                        description=description)
+        degree.save()
+        return degree
+
+    def test_create_degree(self):
+        degree = Degree.create_degree(degree_id=1, name="History",
```
Comment (Yarboa) [inline AcadeMeData/test_degree.py:16]: see you are creating the Degree here. so why do you need the fixture? 

You should call the fixture from test like that test_create_degree(self, generate_degree):

Remove line 16, 
now you have to search for degree in database objects

so you test should be changed.
see this as hoe users tests are going 
user = User.get_user('username')  # search for the user in the db by username
Replace the Degree class and search for the content

https://github.com/redhat-beyond/AcadeMe/blob/main/AcadeMeData/tests.py#L17-L20

Comment (DanielItkes) [reply AcadeMeData/test_degree.py]: 'create_degree' is the fucntion I'm testing so I can't remove that.
Everything else is done.

File: AcadeMeData/test_degree.py [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+    def generate_degree(self, degree_id=1, name='History', universities="Ben Gurion University, Reichman University",
+                        description="Learn about historic events and their influences on the world"):
+        degree = Degree(degree_id=degree_id, name=name, universities=universities,
+                        description=description)
+        degree.save()
+        return degree
+
+    def test_create_degree(self, generate_degree):
+        degree = create_degree(degree_id=1, name='History', universities="Ben Gurion University, Reichman University",
+                               description="Learn about historic events and their influences on the world"))
```
Comment (Yarboa) [inline AcadeMeData/test_degree.py]: Please remove this one
It is generate in the fixture
now you need to search for it and compare to generate_degree.name
Please read the comments, every thing is there

Comment (DanielItkes) [reply AcadeMeData/test_degree.py]: Done

**Review by Yarboa — APPROVED** [review]
LGTM

**Pull request merged**